src/bentoml/_internal/frameworks/transformers.py

```diff
@@ -34,50 +79,53 @@
         "'transformers' is required in order to use module 'bentoml.transformers'. Install transformers with 'pip install transformers'."
     )
 
+if t.TYPE_CHECKING:
+    TransformersPreTrained = (
+        transformers.PreTrainedTokenizerBase
+        | transformers.PreTrainedTokenizer
+        | transformers.PreTrainedTokenizerFast
+        | transformers.PreTrainedModel
+        | transformers.TFPreTrainedModel
+        | transformers.FlaxPreTrainedModel
+        | transformers.image_processing_utils.BaseImageProcessor
+        | transformers.SequenceFeatureExtractor
+        | transformers.Pipeline
+    )
```

Comment:
Should we keep all the type checking code up above? Also, are we going too far with typing support and will we have to maintain this list up-to-date with Transformers? I'm not very familiar with typing so adding @sauyon and @bojiang for their thoughts.

Reply:
This will cover all base classes for all frameworks that extend the PreTrainedModel and Pipeline class. I think having this type is ok?

Reply:
I don't feel strong either way but there are a lot more pre-trained classes than this. It is not feasible to keep the list up-to-date here.